pkg/R/sparkR.R

```diff
@@ -47,14 +34,16 @@ sparkR.stop <- function(env = .sparkREnv) {
   
     callJStatic("SparkRHandler", "stopBackend")
     # Also close the connection and remove it from our env
-    conn <- get(".sparkRCon", env)
+    conn <- get(".sparkRCon", envir = env)
     close(conn)
     rm(".sparkRCon", envir = env)
-    # Finally, sleep for 1 sec to let backend finish exiting.
-    # Without this we get port conflicts in RStudio when we try to 'Restart R'.
-    Sys.sleep(1)
```

Comment:
So it turns out we can't completely get rid of the finalizer because of this case where we try to 'Restart R' from RStudio. (You can reproduce this pretty easily. Just launch RStudio, load SparkR, call sparkR.init() and then press 'Restart R' in the Session menu -- Without the sleep the process will hang).

So the question is do we add a new finalizer which just calls `Sys.sleep(1)` or do we reuse `sparkR.stop` as the finalizer ? I'm in favor of the former option just to keep the finalizer simple.